cc/tflite_micro/tflite_micro.cc

```diff
@@ -93,15 +113,18 @@ int tflite_run(
     return -1;
   }
 
-  if (input_bytes_len != input->bytes) {
+  if (input_bytes_len != input_size) {
     MicroPrintf("Expected input len: %d bytes but got %d bytes\n",
-                input->bytes,
+                input_size,
                 input_bytes_len);
     return -1;
   }
 
   // Place input in the model's input tensors
-  memcpy(input->data.uint8, input_bytes_ptr, input_bytes_len);
+  for (size_t i = 0, offset = 0; i < interpreter->inputs_size(); i++) {
+    memcpy(inputs[i]->data.uint8, input_bytes_ptr + offset, inputs[i]->bytes);
```

Comment:
Why do we need multiple input tensors?
I thought that TFLite models only accept a single tensor as an input?
Or models may have multiple inputs?

Reply:
Right, models can have multiple input tensors.
Think of a text sequence: "dog is running in the yard":
`dog` can be put in one input tensor to realize who or what animal.
`running` can be put in another input tensor to realize behavior.
`yard` can be put in another input tensor to realize venue.

In short, multiple input tensors learn and realize multiple features from the input.